### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by TimothyGu**
Bug fixes:

- #5 
- nodejs/node#10952
- PR URLs with `#` fragments

Also some internal simplifications of RegExps, Promises, Maps, etc.

### CODE REVIEW

File: review.js [hunk 1]
```diff
@@ -120,35 +120,13 @@
     }
   }
 
-  function getFixesUrl() {
```
Comment (evanlucas) [inline review.js:123]: was this not used?

File: review.js [hunk 2]
```diff
@@ -372,50 +349,39 @@
...
+      if (Array.from(paragraphs).some(p => LGTM_RE.test(p.innerHTML))) {
         revs.push(login)
       }
     }
 
     return revs
   }
 
-  function getCollaborators(cb) {
+  function getCollaborators() {
```
Comment (evanlucas) [inline review.js]: I would prefer to keep the callback here vs returning a promise

File: review.js [hunk 3]
```diff
@@ -372,50 +349,39 @@
       const href = a.getAttribute('href')
       if (!href) continue
       const login = href.slice(1).toLowerCase()
-      const body = comment.querySelector('.comment-body')
-      if (body && LGTM_RE.test(body.innerHTML)) {
+      const paragraphs = comment.querySelectorAll('.comment-body > p')
+      if (Array.from(paragraphs).some(p => LGTM_RE.test(p.innerHTML))) {
```
Comment (evanlucas) [inline review.js:355]: just curious, what's the reason for this change? I find this a lot more difficult to read than the previous implementation.

**Review by evanlucas — CHANGES_REQUESTED** [review]
Wow, thanks a lot for the contributions! I left a few comments, but overall looks pretty good.

File: review.js (hunk 1 above)
Comment (TimothyGu) [reply review.js:123]: No it wasn't.

File: review.js (hunk 3 above)
Comment (TimothyGu) [reply review.js:355]: In https://github.com/nodejs/node/pull/10657, a LGTM embedded in a `<blockquote>` (and therefore one that does not belong to the author of the comment) was recognized. This piece of code only looks for direct `<p>` children of `.comment-body`, and mitigates that issue.

Comment (evanlucas) [reply review.js:355]: ah cool that makes sense.

File: review.js (hunk 2 above)
Comment (TimothyGu) [reply review.js]: Okay, but I wonder what the reason for this is?

Comment (evanlucas) [reply review.js]: It is strictly preference. I prefer nodebacks over promises. :]

Comment (TimothyGu) [reply review.js]: Fixed. PTAL!

**Review by evanlucas — APPROVED** [review]

**Comment by evanlucas** [comment]
Thanks!

**Pull request merged**